diode-server/dbstore/postgres/repository.go

```diff
@@ -56,6 +56,19 @@ func (r *Repository) CreateIngestionLog(ctx context.Context, ingestionLog *recon
 	return &createdIngestionLog.ID, nil
 }
 
+// RetrieveIngestionLogByExternalID looks up an ingestion log using its external identifier (uuid)
+func (r *Repository) RetrieveIngestionLogByExternalID(ctx context.Context, uuid string) (*int32, *reconcilerpb.IngestionLog, error) {
+	ingestionLog, err := r.queries.RetrieveIngestionLogByExternalID(ctx, uuid)
+	if err != nil {
+		return nil, nil, err
+	}
+	log, err := ingestionLog.ToProtobuf()
```

Comment:
As in the comment for adapters:
```suggestion
	log, err := ingestionLog.ToProto()
```